fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/utils/ChangelogRowConverter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.connector.flink.utils;
+
+import com.alibaba.fluss.client.table.scanner.ScanRecord;
+import com.alibaba.fluss.types.RowType;
+
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+
+public class ChangelogRowConverter extends FlussRowToFlinkRowConverter {
```

Comment:
It would be better to use the **composite design pattern** instead of extending in this case, as `ChangelogRowConverter` does not represent a direct row-to-row conversion.  

We can introduce a pure `RecordToFlinkRowConverter` interface to abstract the conversion of `ScanRecord` to `RowData`. This interface would have two implementations:  
1. One implementation (e.g., `DirectRowConverter` or `PlainRowConverter`) directly invokes `FlussRowToFlinkRowConverter#toFlinkRowData(ScanRecord)` for the conversion.  
2. Another implementation (e.g., `ChangelogRowConverter`) wraps `FlussRowToFlinkRowConverter` to generate the `baseRowData` and enrich it with additional metadata columns.